api/restHandler/WebhookListenerGithubImpl.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020 Devtron Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package restHandler
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	client "github.com/devtron-labs/devtron/client/events"
+	"github.com/devtron-labs/devtron/client/pubsub"
+	"github.com/devtron-labs/devtron/pkg/pipeline"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type WebhookListenerGithubImpl struct {
+	logger *zap.SugaredLogger
+	gitHostConfig    	 pipeline.GitHostConfig
+	eventClient          client.EventClient
+}
+
+func NewWebhookListenerGithubImpl(logger *zap.SugaredLogger, gitHostConfig pipeline.GitHostConfig,
+	eventClient client.EventClient) *WebhookListenerGithubImpl {
+	return &WebhookListenerGithubImpl{
+		logger: logger,
+		gitHostConfig: gitHostConfig,
+		eventClient: eventClient,
+	}
+}
+
+func (impl WebhookListenerGithubImpl) OnWebhookEvent(w http.ResponseWriter, r *http.Request) {
+	impl.logger.Debug("github webhook event came")
+
+	// validate secret
+	secret, err := impl.gitHostConfig.GetGitHostSecretByName(pipeline.GIT_HOST_NAME_GITHUB)
```

Comment:
are we using same secret for all webhook events from github? this is security issue, we should use separate one for each.. reduces the impact surface area and also easier to revoke and regenerate

Reply:
Secret is at globel level of gitHost